File: Cloude/Cloude/Utilities/Colors.swift
import SwiftUI

extension Color {
    static var themeBackground: Color { Color(hex: AppTheme.current.palette.background) }
    static var themeSecondary: Color { Color(hex: AppTheme.current.palette.secondary) }
    static func themeBackground(_ theme: AppTheme) -> Color { Color(hex: theme.palette.background) }
    static func themeSecondary(_ theme: AppTheme) -> Color { Color(hex: theme.palette.secondary) }

    static let pastelGreen = Color(hex: 0x7AB87A)
    static let pastelRed = Color(hex: 0xB54E5E)

    static func fromName(_ name: String?, default defaultColor: Color = .blue) -> Color {
        switch name?.lowercased() {
        case "blue": return .blue
        case "green": return .green
        case "red": return .red
        case "purple": return .purple
        case "orange": return .orange
        case "cyan": return .cyan
        case "magenta", "pink": return .pink
        case "yellow": return .yellow
        case "teal": return .teal
        case "indigo": return .indigo
        case "mint": return .mint
        default: return defaultColor
        }
    }

    init(hex: UInt, alpha: Double = 1.0) {
        self.init(
            .sRGB,
            red: Double((hex >> 16) & 0xff) / 255,
            green: Double((hex >> 8) & 0xff) / 255,
            blue: Double(hex & 0xff) / 255,
            opacity: alpha
        )
    }
}

File: android/app/src/main/java/com/cloude/app/Utilities/Colors.kt
package com.cloude.app.Utilities

import androidx.compose.ui.graphics.Color

val Accent = Color(0xFFCC7257)

val PastelGreen = Color(0xFF7AB87A)
val PastelRed = Color(0xFFB54E5E)

fun Color.Companion.fromHex(hex: Long): Color = Color(hex)

fun Color.Companion.fromName(name: String?, default: Color = Color.Blue): Color =
    when (name?.lowercase()) {
        "blue" -> Color.Blue
        "green" -> Color.Green
        "red" -> Color.Red
        "purple" -> Color(0xFF9C27B0)
        "orange" -> Color(0xFFFF9800)
        "cyan" -> Color.Cyan
        "magenta", "pink" -> Color(0xFFE91E63)
        "yellow" -> Color.Yellow
        "teal" -> Color(0xFF009688)
        "indigo" -> Color(0xFF3F51B5)
        "mint" -> Color(0xFF00BFA5)
        else -> default
    }
